src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -76,6 +64,32 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
     result
   }
 
+  private def prepareTransactions(transactions: Seq[SignedTransaction], blockGasLimit: BigInt) = {
+    val sortedTransactions = transactions.groupBy(_.senderAddress).values.toList.flatMap { txsFromSender =>
+      val ordered = txsFromSender
+        .sortBy(-_.tx.gasPrice)
+        .sortBy(_.tx.nonce)
+        .scanLeft(None: Option[BigInt], None: Option[SignedTransaction]) { case ((prevNonce, _), stx) =>
```

Comment:
I would go with
```
.foldLeft(Seq.empty[SignedTransaction]) { case (txs, tx) =>
  if (txs.exists(_.tx.nonce == tx.tx.nonce)) {
    txs
  } else {
    tx +: txs
  }
}
.reverse
```
to not append to Seq since it defaults to List
do you think it will be easier to read?